frontend/src/components/ImplementationReportView/ImplementationReportView.tsx

```diff
@@ -0,0 +1,121 @@
+import { Card } from "react-bootstrap";
+import { useLoaderData, useParams, Link } from "react-router-dom";
+import { Table, Container } from "react-bootstrap";
+import { ImplementationMetadata } from "../../data/parseReportData";
+// @ts-ignore
+import LoadingAnimation from "../LoadingAnimation";
+import DialectCompliance from "./DialectCompliance";
+import { mapLanguage } from "../../data/mapLanguage";
+
+export const ImplementationReportView = () => {
+  const allImplementations = useLoaderData() as {
+    [key: string]: ImplementationMetadata;
+  };
+
+  const { langImplementation } = useParams();
+  const implementationName = langImplementation ?? "";
+
+  function filterImplementation(
```

Comment:
I have to stare at this longer to understand what it does, but it seems a bit like it's doing a somewhat odd kind of pivot (taking data keyed by something and pivoting it to be keyed by something else).

In general it'd be easier to follow some of these if you added a few docstrings I think on what these are trying to do -- the function name could be a bit clearer too -- this is filtering what, implementation metadata? To what? Does `ImplementationMetadata` represent *all* implementations' metadata? And this filters to just one specific implementation? (That sounds a bit odd to me, I'd instead expect not to have an object representing *all* metadata, to instead simply have that be a mapping / JS object from implementation ID to an object representing *one* implementation's metadata. But perhaps that isn't something new to this PR.)

Reply:
> This is filtering what, implementation metadata?

Yes, it returns metadata of the page-specific implementation.

> Does ImplementationMetadata represent all implementations' metadata?

Nope. It represents just one specific implementation.

> I'd instead expect not to have an object representing all metadata, but instead, to simply have that be a mapping / JS object from implementation ID to an object representing one implementation's metadata.

The current structure represents the same thing as you mentioned above.

See this:  
![image](https://github.com/bowtie-json-schema/bowtie/assets/113545196/de6d2649-29ae-405f-b237-944223a8b081)

Reply:
I still don't really understand then what / why we're iterating over `Object.keys` and filtering that, that's a code smell. Why isn't this simply a key-value lookup? Why iterate over a bunch of keys trying to find the implementation we're looking for?